server/memphis_handlers_functions.go

```diff
@@ -0,0 +1,222 @@
+			c.AbortWithStatusJSON(SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
+			return
+		} else {
+			if integration, ok := tenantIntegrations[k].(models.Integration); ok {
+				connectedRepos, err := fh.GetConnectedSourceCodeRepos(integration)
+				if err != nil {
+					serv.Errorf("[tenant: %v][user: %v]GetAllFunctions at GetConnectedSourceCodeRepos: %v", user.TenantName, user.Username, err.Error())
+					continue
+				}
+				var repoContent interface{}
+				for _, connectedRepo := range connectedRepos {
+					connectedRepoRes := connectedRepo.(map[string]interface{})
+					repo := connectedRepoRes["repo_name"].(string)
+					owner := connectedRepoRes["repo_owner"].(string)
+					repoContent, err = fh.GetContentOfSelectedRepos(repo, owner, k, integration)
+					if err != nil {
+						serv.Errorf("[tenant: %v][user: %v]GetAllFunctions at GetContentOfSelectedRepos: %v", user.TenantName, user.Username, err.Error())
+						continue
+					}
+					functions, err = GetFunctionsDetails(user, k, integration, repoContent, connectedRepoRes, functions)
+					if err != nil {
+						serv.Errorf("[tenant: %v][user: %v]GetAllFunctions at GetFunctionsDetails: %v", user.TenantName, user.Username, err.Error())
+						continue
+					}
+				}
+			}
+		}
+	}
+	c.JSON(200, functions)
+}
+
+func validateYamlContent(yamlMap map[string]interface{}) error {
+	requiredFields := []string{"function_name", "description", "tags", "language"}
+	missingFields := make([]string, 0)
+	for _, field := range requiredFields {
+		if _, exists := yamlMap[field]; !exists {
+			missingFields = append(missingFields, field)
+		}
+	}
+
+	if len(missingFields) > 0 {
+		return fmt.Errorf("Missing fields: %v\n", missingFields)
+	}
+	return nil
+}
+
+func (fh FunctionsHandler) GetConnectedSourceCodeRepos(integration models.Integration) ([]interface{}, error) {
+	connectedRepos := integration.Keys["connected_repos"].([]interface{})
+	selectedRepos := []interface{}{}
+	for _, repo := range connectedRepos {
+		repository := repo.(map[string]interface{})
+		repoType := repository["type"].(string)
+		if repoType == "functions" {
+			selectedRepos = append(selectedRepos, repo)
+		}
+	}
+	return selectedRepos, nil
+}
+
+func (fh FunctionsHandler) GetContentOfSelectedRepos(repo, owner string, sourceCodeIntegrationType string, integration models.Integration) ([]*github.RepositoryContent, error) {
+	repos, err := GetContentDetails(repo, owner, sourceCodeIntegrationType, integration)
+	if err != nil {
+		return []*github.RepositoryContent{}, err
+	}
+	return repos, nil
+}
+
+func GetFunctionsDetails(user models.User, sourceCodeTypeIntegration string, integration models.Integration, repoContent interface{}, connectedRepo map[string]interface{}, functions []FunctionsResult) ([]FunctionsResult, error) {
+	repo := connectedRepo["repo_name"].(string)
+	owner := connectedRepo["repo_owner"].(string)
+	branch := connectedRepo["branch"].(string)
```

Comment:
Not good this is exactly what I told you that should sit in the scam.go if tomorrow we will have another integrations I don’t want to maintain itoverhere